File: src/0067-Add-Binary/0067.cpp
class Solution 
{
public:
    string addBinary(string a, string b) 
    {
        string res;
        int carry = 0, l1 = a.size(), l2 = b.size();
        while (carry or l1 or l2)
        {
            if (l1) carry += a[--l1] - '0';
            if (l2) carry += b[--l2] - '0';
            res.push_back(carry%2 + '0');
            carry >>= 1;
        }
        reverse(res.begin(), res.end());
        return res;
    }
};

File: src/0067-Add-Binary/0067.py
class Solution:
    def addBinary(self, a, b):
        """
        :type a: str
        :type b: str
        :rtype: str
        """
        res = list()
        add, l1, l2 = 0, len(a), len(b)
        while add or l1 or l2:
            if l1:
                l1 -= 1
                add += int(a[l1])
            if l2:
                l2 -= 1
                add += int(b[l2])
            res.append(str(add & 1))
            add >>= 1
            
        return ''.join(reversed(res))